harness/determined/core/_train.py

```diff
@@ -80,28 +82,46 @@ def report_training_metrics(
         but may be accessed from the master using the CLI for post-processing.
         """
 
-        body = {
-            "trial_run_id": self._run_id,
-            "steps_completed": steps_completed,
-            "metrics": {
-                "avg_metrics": metrics,
-            },
-        }
-        if batch_metrics is not None:
-            body["metrics"]["batch_metrics"] = batch_metrics  # type: ignore
-        logger.info(
-            f"report_training_metrics(steps_completed={steps_completed}, metrics={metrics})"
-        )
-        self._session.post(
-            f"/api/v1/trials/{self._trial_id}/training_metrics",
-            data=det.util.json_encode(body),
+        # CHECK: do we want this for training as well? same for batch_metrics?
+        serializable_metrics = self._get_serializable_metrics(metrics)
+        reportable_metrics = {k: metrics[k] for k in serializable_metrics}
+
+        v1metrics = bindings.v1Metrics(avgMetrics=reportable_metrics, batchMetrics=batch_metrics)
+        v1TrialMetrics = bindings.v1TrialMetrics(
+            metrics=v1metrics,
+            stepsCompleted=total_batches,
+            trialId=self._trial_id,
+            trialRunId=self._run_id,
         )
+        body = bindings.v1ReportTrialMetricsRequest(metrics=v1TrialMetrics, type=metric_type)
+        bindings.post_ReportTrialMetrics(self._session, body=body, metrics_trialId=self._trial_id)
 
+        # Also sync tensorboard (all metrics, not just json-serializable ones).
         if self._tensorboard_mode == TensorboardMode.AUTO:
             if self._tbd_writer:
-                self._tbd_writer.on_train_step_end(steps_completed, metrics, batch_metrics)
+                if metric_type == LEGACY_VALIDATION:
+                    self._tbd_writer.on_validation_step_end(total_batches, metrics)
+                elif metric_type == LEGACY_TRAINING:
+                    self._tbd_writer.on_train_step_end(total_batches, metrics, batch_metrics)
+                else:
+                    pass  # FIXME
             self._tensorboard_manager.sync()
 
+    def report_training_metrics(
```

Comment:
should this be deprecated?

Reply:
they don't have to be deprecated but we could deprecate them. I updated it.